### DISCUSSION TITLE
Where is stored which settings and storage account to use?

### DISCUSSION THREAD

**Discussion by jurjenoskam** [Azure/CloudShell#139] [Q&A] [answered]
I have tried to deploy Cloud Shell in a VNet according to [the documentation](https://docs.microsoft.com/en-us/azure/cloud-shell/private-vnet), but have not been successful. When I try to start the Cloud Shell from the Portal, I get the following error:

```
Requesting a Cloud Shell in virtual network. This may take several minutes.Failed to provision a Cloud Shell: {"code":"InternalServerError","message":"Encountered an internal server error. The tracking activity id is '86b470f1-c9bb-4941-ab90-ae5778d97d70', correlation id is 'b6e2d69c-3823-4bde-911d-1d0ef3b608de'."}
```

My question isn't about this error (I have opened a support ticket for that). I'd like to keep the Storage Account and related resources (VNet, relay, etc) in place, but would also like to reconfigure Cloud Shell to use another Storage Account and not run it in a VNet for now. I know that I can normally achieve this by running the command `clouddrive unmount`, but I can't run that command since the shell won't start. Looking in the Storage Account, there is nothing in the file share, nor are there any blobs (apart from `$logs`), queues or tables.

Where is the configuration kept about how Cloud Shell should be started, and how can I influence that without having access to a shell session (and without having to delete the storage account or any other resource)?

**Comment by edyoung** [comment, marked as answer]
The settings are stored internally by CLoud Shell and accessed via REST API. 
https://docs.microsoft.com/en-us/azure/cloud-shell/troubleshooting#delete provides some steps to delete them from outside the shell.

**Reply by jurjenoskam** [reply]
I read that page _twice_ and managed to overlook that segment both times, apparently... Thanks for pointing me towards the answer!

**Reply by matarillo** [reply]
Hi @edyoung, 

I tried it via pwsh and got an error.

```
PS C:\Users\me> $token = (Get-AzAccessToken -Resource  https://management.azure.com/).Token
PS C:\Users\me> Invoke-WebRequest -Method Delete -Uri https://management.azure.com/providers/Microsoft.Portal/usersettings/cloudconsole?api-version=2017-12-01-preview -Headers @{Authorization = "Bearer $token"}
Invoke-WebRequest: {
  "error": {
    "code": "ClientNotAllowed",
    "message": "The client with application Id '1950a258-227b-4e31-a9cf-717495945fc2' is not permitted to access this API."
  }
}
```

ID `1950a258-227b-4e31-a9cf-717495945fc2` is for Azure PowerShell, AFAIK.
Should I use Azure CLI on bash?

**Reply by edyoung** [reply]
That's odd. I wasn't able to reproduce the error you saw, but I did get a message "The request token does not have 'puid' claim. This user type is not supported." . Not sure if that is an issue in Azure Powershell or something on my computer.

I was able to use az cli successfully (that can also be run from powershell btw if that is more convenient for you) so that seems like a workaround worth trying.
